### DISCUSSION TITLE
Could not load the Qt platform, macOS Docker

### DISCUSSION THREAD

**Discussion by lustefan97** [The-OpenROAD-Project/OpenROAD-flow-scripts#1515] [Q&A] [answered]
Hello,

I have recently acquired a 2019 Intel MBP on which I would like to install OpenROAD (using Ubuntu 22.04 in Docker). 
I followed this tutorial [https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithDocker.html](url) and even though openroad seems to be installed (compiled and can open it using the `openroad` command, I haven't tried to run a design yet) I can not access the GUI either trough `openroad -gui` or `gui::show`. The logs that I obtain are as follows : 

``` 
openroad> gui::show
[WARNING GUI-0076] could not connect to display 192.168.1.166:0
[INFO GUI-0075] Could not load the Qt platform plugin "xcb" in "" even though it was found.
[ERROR GUI-0077] This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

terminate called after throwing an instance of 'std::runtime_error'
  what():  GUI-0077
Signal 6 received
Stack trace:
 0# 0x000055E8A0690DAB in openroad
 1# 0x00007FD849FEF520 in /lib/x86_64-linux-gnu/libc.so.6
 2# pthread_kill in /lib/x86_64-linux-gnu/libc.so.6
 3# raise in /lib/x86_64-linux-gnu/libc.so.6
 4# abort in /lib/x86_64-linux-gnu/libc.so.6
 5# 0x00007FD84A380B9E in /lib/x86_64-linux-gnu/libstdc++.so.6
 6# 0x00007FD84A38C20C in /lib/x86_64-linux-gnu/libstdc++.so.6
 7# 0x00007FD84A38B1E9 in /lib/x86_64-linux-gnu/libstdc++.so.6
 8# __gxx_personality_v0 in /lib/x86_64-linux-gnu/libstdc++.so.6
 9# 0x00007FD84A1EB884 in /lib/x86_64-linux-gnu/libgcc_s.so.1
10# _Unwind_Resume in /lib/x86_64-linux-gnu/libgcc_s.so.1
11# 0x00007FD84AB712F9 in /lib/x86_64-linux-gnu/libQt5Core.so.5
12# QMessageLogger::fatal(char const*, ...) const in /lib/x86_64-linux-gnu/libQt5Core.so.5
13# QGuiApplicationPrivate::createPlatformIntegration() in /lib/x86_64-linux-gnu/libQt5Gui.so.5
14# QGuiApplicationPrivate::createEventDispatcher() in /lib/x86_64-linux-gnu/libQt5Gui.so.5
15# QCoreApplicationPrivate::init() in /lib/x86_64-linux-gnu/libQt5Core.so.5
16# QGuiApplicationPrivate::init() in /lib/x86_64-linux-gnu/libQt5Gui.so.5
17# QApplicationPrivate::init() in /lib/x86_64-linux-gnu/libQt5Widgets.so.5
18# gui::startGui(int&, char**, Tcl_Interp*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) in openroad
19# show(char const*, bool) in openroad
20# 0x000055E8A0D80AF5 in openroad
21# TclNRRunCallbacks in /lib/x86_64-linux-gnu/libtcl8.6.so
22# 0x00007FD84DE87028 in /lib/x86_64-linux-gnu/libtcl8.6.so
23# Tcl_EvalEx in /lib/x86_64-linux-gnu/libtcl8.6.so
24# Tcl_Eval in /lib/x86_64-linux-gnu/libtcl8.6.so
25# ord::tclAppInit(Tcl_Interp*) in openroad
26# Tcl_MainEx in /lib/x86_64-linux-gnu/libtcl8.6.so
27# main in openroad
28# 0x00007FD849FD6D90 in /lib/x86_64-linux-gnu/libc.so.6
29# __libc_start_main in /lib/x86_64-linux-gnu/libc.so.6
30# _start in openroad
Aborted
```

Problem is that I checked and the QT library is indeed installed, tried again to `apt-get update` to no effect and installed xcb again to no effect. I did followed the tutorial to enable gui app in docker for macos, I installed socat and xquartz and managed to have a xeyes gui app running. I also ran the setup.sh script and the DependencyInstaller.sh script....

I don't really know what could be the issue here and how to solve it...

**Comment by maliberty** [comment]
@eder-matheus any ideas?

**Reply by eder-matheus** [reply]
I've never used OpenROAD with Docker on my Mac, but I'll try it and see if I can make it work. I know that the default Qt in MacOS brew is Qt6, while we use Qt5 in OR. But maybe it is not related, since it generates a compile error, not a crash.

**Comment by vvbandeira** [comment]
@lustefan97 
Did you follow the instructions about `Enable GUI`:
https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithDocker.html#enable-gui-support
And the external tutorial we suggest:
https://cntnr.io/running-guis-with-docker-on-mac-os-x-a14df6a76efc
Were you able to open Chrome as per the tutorial?

**Reply by lustefan97** [reply]
I did follow the tutorial, but I must admit that I didn't try to open chrome. I only ran the xeyes app.

One question that I have though is that I removed the -rm from the last _docker run_ command in order to be able to keep the container with the permission. Is it a mistake, should I keep an Ubuntu image with OpenROAD and run the command each time ?

**Reply by eder-matheus** [reply, marked as answer]
@lustefan97 I was able to open the GUI by initializing Docker with the following command line:
```docker run --rm -it -u $(id -u ${USER}):$(id -g ${USER}) **-e DISPLAY=YOUT_IP:0** -v $(pwd)/flow:/OpenROAD-flow-scripts/flow openroad/flow-ubuntu22.04-builder.```

Did you add the `-e` option when initializing the docker image?

EDIT: Notice that you have to keep the `socat TCP-LISTEN:6000,reuseaddr,fork UNIX-CLIENT:\"$DISPLAY\"` in a parallel terminal.

**Reply by lustefan97** [reply]
@eder-matheus I was able to access the guy too just now. I made two mistakes : the first being not keeping `socat TCP-LISTEN:6000,reuseaddr,fork UNIX-CLIENT:\"$DISPLAY\"` in a parallel terminal, the second being on the image used. I had to build a docker image from the ubuntu container in which OpenROAD was built in order to use it in the _docker run_ command. First time docker user, but now it's working ! Thanks for your help !
